src/integration_tests/helpers/integration_helpers.py

```diff
@@ -0,0 +1,284 @@
+import json
+import os
+import shutil
+import time
+from pathlib import Path
+
+import firebase_admin
+import google.auth.transport.requests
+import google.oauth2.id_token
+import requests
+from config.config_factory import ConfigFactory
+from firebase_admin import _apps, firestore
+from google.cloud import storage
+from requests.adapters import HTTPAdapter
+from urllib3 import Retry
+
+config = ConfigFactory.get_config()
+
+if not _apps:
+    firebase_admin.initialize_app()
+
+db = firestore.client()
+storage_client = storage.Client()
+
+
+def setup_session() -> requests.Session:
+    """
+    Method to setup a http/s session to facilitate testing.
+
+    Parameters:
+        None
+
+    Returns:
+        Session: a http/s session.
+    """
+    session = requests.Session()
+    retry = Retry(connect=3, backoff_factor=0.5)
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+
+    return session
+
+
+def generate_headers() -> dict[str, str]:
+    """
+    Method to create headers for authentication if connecting to a remote version of the API.
+
+    Parameters:
+        None
+
+    Returns:
+        dict[str, str]: the headers required for remote authentication.
+    """
+    headers = {}
+    auth_token = os.environ.get("ACCESS_TOKEN")
+    if auth_token is None:
+        auth_req = google.auth.transport.requests.Request()
+        auth_token = google.oauth2.id_token.fetch_id_token(auth_req, config.API_URL)
+
+    headers = {"Authorization": f"Bearer {auth_token}"}
+
+    return headers
+
+
+def load_json(filepath: str) -> dict:
+    """
+    Method to load json from a file.
+
+    Parameters:
+        filepath: string specifiing the location of the file to be loaded.
+
+    Returns:
+        dict: the json object from the specifiede file.
+    """
+    with open(filepath) as f:
+        return json.load(f)
+
+
+def create_dataset(
+    filename: str, dataset: dict, session: requests.Session, headers: dict[str, str]
+) -> int | None:
+    """
+    Method to create a dataset using either the remote new dataset function or the local version.
+
+    Parameters:
+        filename: the filename to use for the dataset
+        dataset: the dataset to be created
+        session: a session instance for http/s connections
+        headers: the relevant headers for authentication for http/s calls
+
+    Returns:
+        int | None: status code for local function and no return for remote.
+    """
+    if config.API_URL.__contains__("local"):
```

Comment:
I will suggest to refractor this logic in distinguishing a local test from remote test into a specific function within helpers

Reply:
Personally im not sure theres value in that, it would be a one line function that would make the flow of the helpers more complicated to follow